core/src/main/java/org/apache/iceberg/MetadataUpdateParser.java

```diff
@@ -234,6 +290,58 @@ private static void writeAddSortOrder(MetadataUpdate.AddSortOrder update, JsonGe
     SortOrderParser.toJson(update.sortOrder(), gen);
   }
 
+  private static void writeSetDefaultSortOrder(MetadataUpdate.SetDefaultSortOrder update, JsonGenerator gen)
+      throws IOException {
+    gen.writeNumberField(SORT_ORDER_ID, update.sortOrderId());
+  }
+
+  private static void writeAddSnapshot(MetadataUpdate.AddSnapshot update, JsonGenerator gen) throws IOException {
+    gen.writeFieldName(SNAPSHOT);
+    SnapshotParser.toJson(update.snapshot(), gen);
+  }
+
+  // TODO - Reconcile the spec's set-based removal with the current class implementation that only handles one value.
+  private static void writeRemoveSnapshots(MetadataUpdate.RemoveSnapshot update, JsonGenerator gen) throws IOException {
+    gen.writeArrayFieldStart(SNAPSHOT_IDS);
+    for (long snapshotId : ImmutableSet.of(update.snapshotId())) {
+      gen.writeNumber(snapshotId);
+    }
+    gen.writeEndArray();
+  }
+
+  private static void writeSetSnapshotRef(MetadataUpdate.SetSnapshotRef update, JsonGenerator gen) throws IOException {
+    gen.writeStringField(REF_NAME, update.name());
+    gen.writeNumberField(SNAPSHOT_ID, update.snapshotId());
+    gen.writeStringField(TYPE, update.type());
+    // TODO - Should we explicitly set the null values?
+    JsonUtil.writeIntegerFieldIf(update.minSnapshotsToKeep() != null,
+        MIN_SNAPSHOTS_TO_KEEP, update.minSnapshotsToKeep(), gen);
+    JsonUtil.writeLongFieldIf(update.maxSnapshotAgeMs() != null,
+        MAX_SNAPSHOT_AGE_MS, update.maxSnapshotAgeMs(), gen);
+    JsonUtil.writeLongFieldIf(update.maxRefAgeMs() != null, MAX_REF_AGE_MS,
+        update.maxRefAgeMs(), gen);
```

Comment:
Question: Should we explicitly set the `null` values or not?

Reply:
I think they should be omitted.

Reply:
The line for `MIN_SNAPSHOTS_TO_KEEP` can't fit on one line. Everything else seems like it can.

Updating.

Reply:
Updated to put as many on one line as possible.